.gitignore

```diff
@@ -6,5 +6,10 @@ build-storybook.log
 .env
 .npmrc
 
-src/**/*.js
-src/**/*.js.map
\ No newline at end of file
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
+.pnp.*
```

Comment:
Let me know if I've made mistakes here as I'm not familiar with Yarn 2-4. I have used the recommendations documented by the Yarn team.

I've removed the *js gitignores because I've made `tsc` skip the JS file emitting phase. Vite can take care of that during the build process.